### ISSUE TITLE
Nested items aren't being translated to their correct types

### ISSUE THREAD

**Issue by jordanwalsh23** [XeroAPI/xero-node#13]
Currently for objects that have a nested schema (e.g. TaxRate -> TaxComponent, or Invoice -> Payments), the nested schema isn't being translated from the XML into it's appropriate type.

e.g. 

```
//Tax Component Schema
var TaxComponentSchema = new Entity.SchemaObject({
    Name: { type: String, toObject: 'always' },
    Rate: { type: String, toObject: 'always' },
    IsCompound: { type: Boolean }
});

//Get TaxComponents Test
currentApp.core.taxrates.getTaxRates()
    .then(function(taxRates) {
        var taxRate = taxRates[0];
        _.each(taxRate.TaxComponents, function(taxComponent) {
                //This test fails as taxComponent.IsCompound is a 'String' not a 'Boolean'.
                expect(taxComponent.IsCompound).to.be.oneOf([true, false]);
        });
    });
```

Fix is to ensure that the marshalling respects the types for nested objects.

**Comment by jordanwalsh23** [comment]
Fixed the 'Boolean' issue as part of #46.  

However the number to string conversion is an issue as the default libraries will knock off leading zeroes.  This works well for fields like TaxRate, but poorly for fields like Phone Area Code (e.g. `03` should be retained).

Still investigating here.

**Comment by jordanwalsh23** [comment]
Resolved as of xero-node@2.5.0
